packages/iris-grid/src/IrisGridPartitionedTableModel.ts

```diff
@@ -0,0 +1,442 @@
+/* eslint class-methods-use-this: "off" */
+import {
+  DataBarOptions,
+  GridRange,
+  ModelIndex,
+  MoveOperation,
+  VisibleIndex,
+} from '@deephaven/grid';
+import type {
+  Column,
+  ColumnStatistics,
+  CustomColumn,
+  dh as DhType,
+  FilterCondition,
+  Format,
+  LayoutHints,
+  PartitionedTable,
+  RollupConfig,
+  Row,
+  Sort,
+  Table,
+  ValueTypeUnion,
+} from '@deephaven/jsapi-types';
+import { Formatter } from '@deephaven/jsapi-utils';
+import Log from '@deephaven/log';
+import {
+  ColumnName,
+  PendingDataMap,
+  PendingDataErrorMap,
+  UITotalsTableConfig,
+} from './CommonTypes';
+import ColumnHeaderGroup from './ColumnHeaderGroup';
+import IrisGridModel from './IrisGridModel';
+import IrisGridTableModel from './IrisGridTableModel';
+import EmptyIrisGridModel from './EmptyIrisGridModel';
+import { IrisGridThemeType } from './IrisGridTheme';
+
+const log = Log.module('IrisGridPartitionedTableModel');
+
+export function isIrisGridPartitionedTableModel(
+  model: IrisGridModel
+): model is IrisGridPartitionedTableModel {
+  return (
+    (model as IrisGridPartitionedTableModel).partitionedTable !== undefined
+  );
+}
```

Comment:
One thing that I think will help clean this up a bit... 
- Define the interface for a partitioned grid model in a file `PartitionedGridModel.ts`. See `ExpandableGridModel` as an example (though you still want it in the `iris-grid` package).
- Have `IrisGridPartitionedTableModel` and `IrisGridProxyModel` add the `implements PartitionedGridModel` to the class definition
  - That way if there's ever new methods added to PartitionedGridModel, they'll both throw an error
  - Side note - IrisGridTreeTableModel and IrisGridProxyModel should also `implements ExpandableGridModel` as well